core/src/main/java/fr/sncf/osrd/envelope_sim/EnvelopeSimContext.java

```diff
@@ -23,19 +23,27 @@ public EnvelopeSimContext(
         this.tractiveEffortCurveMap = tractiveEffortCurveMap;
     }
 
+    /** Computes the rolling stock effort curves that will be used and creates a context */
+    public static EnvelopeSimContext build(
+            RollingStock rollingStock,
+            PhysicsPath path,
+            double timeStep,
+            Comfort comfort,
+            boolean ignoreElectricalProfiles
+    ) {
+        var curvesAndConditions = rollingStock.mapTractiveEffortCurves(
+                path.getModeAndProfileMap(ignoreElectricalProfiles ? null : rollingStock.powerClass), comfort,
+                path.getLength());
```

Comment:
Nitpick, that expression is getting pretty long and complex, I'd prefer if it's split over several lines and variables